### ISSUE TITLE
Why are my test results far better than the paper results?

### ISSUE THREAD

**Issue by hexiangdong2020** [SMARTlab-Purdue/Husformer#8]
I used Husformer to test it on the WESAD dataset and got the final result (Acc: 91.02%, F1: 87.38%). The results of this test are much higher than those tested on the WESAD dataset in the paper (Acc: 78.68%, F1: 79.51%). Why is this happening? Is it because the paper uses a different metric than the open source code? @zdz0086 @SMARTlab-Purdue @wonsu0513 @R7-Robot 
![image](https://github.com/SMARTlab-Purdue/Husformer/assets/127718414/e266de14-18a5-4bac-b19f-f3fcb1c52a0f)

**Comment by xnes** [comment]
Hello @hexiangdong2020 , how did you make to run the code? I am trying to use cogload but I'm lost with the bugs. 

Did you have to do a lot of changes?

**Comment by Marfbin** [comment]
Hi, @hexiangdong2020 , I'm having trouble replicating the husformer code using the DEAP dataset. The definition of output_dim=1 in the main.py file causes the final output shape of the model to look like this (1024, 1), which causes problems in focalloss for calculating log_softmax for the first dimension, which results in logpt values being all zeros. How did you solve this problem？
![image](https://github.com/user-attachments/assets/71ae2bb4-d77e-408f-b45e-18432bd69430)
![image](https://github.com/user-attachments/assets/e2de112e-2ec3-4bc9-8247-b713c5688c2d)
![image](https://github.com/user-attachments/assets/a0af0f8f-8bc8-4753-bf33-006e946801de)

**Comment by Shahtaj-Khalid** [comment]
> I used Husformer to test it on the WESAD dataset and got the final result (Acc: 91.02%, F1: 87.38%). The results of this test are much higher than those tested on the WESAD dataset in the paper (Acc: 78.68%, F1: 79.51%). Why is this happening? Is it because the paper uses a different metric than the open source code? @zdz0086 @SMARTlab-Purdue @wonsu0513 @R7-Robot ![image](https://private-user-images.githubusercontent.com/127718414/325459498-e266de14-18a5-4bac-b19f-f3fcb1c52a0f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM1MzIwMTQsIm5iZiI6MTczMzUzMTcxNCwicGF0aCI6Ii8xMjc3MTg0MTQvMzI1NDU5NDk4LWUyNjZkZTE0LTE4YTUtNGJhYy1iMTlmLWYzZmNiMWM1MmEwZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIwN1QwMDM1MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03OTgwOGU5MWQ2NGY0YjU3OWU4YzZkZTQxMTE4YTViOWE4ZTZkZTdiY2IzMDZhMGY3ZjNjOGZjM2FmZTU1Y2YxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.IX7UDdOL-bTsua2pq2E_TI_dlfX6qvAKYeJAghn1SxQ)

Hi, is it possible for you to upload your test code? I'm trying run Husformer for WESAD too but facing issues.

**Comment by Bluemax666** [comment]
> I used Husformer to test it on the WESAD dataset and got the final result (Acc: 91.02%, F1: 87.38%). The results of this test are much higher than those tested on the WESAD dataset in the paper (Acc: 78.68%, F1: 79.51%). Why is this happening? Is it because the paper uses a different metric than the open source code?

In the paper they say that the metrics used are average multi-class accuracy (Acc) and multiclass average F1-score (F1). But in the code its just the accuracy and a weird f1 score that outputs a different value than sklearn.metrics.f1_score. The basic accuracy is higher than the average multiclass accuracy because the number of samples of each class is not balanced, and the less represented classes count less in the accuracy than in the average multiclass accuracy.

**Comment by kuanyshbakytuly** [comment]
> Hi, [@hexiangdong2020](https://github.com/hexiangdong2020) , I'm having trouble replicating the husformer code using the DEAP dataset. The definition of output_dim=1 in the main.py file causes the final output shape of the model to look like this (1024, 1), which causes problems in focalloss for calculating log_softmax for the first dimension, which results in logpt values being all zeros. How did you solve this problem？ ![image](https://github.com/user-attachments/assets/71ae2bb4-d77e-408f-b45e-18432bd69430) ![image](https://github.com/user-attachments/assets/e2de112e-2ec3-4bc9-8247-b713c5688c2d) ![image](https://github.com/user-attachments/assets/a0af0f8f-8bc8-4753-bf33-006e946801de)

hi, could you share with remake_label() function?

**Comment by zhangning105** [comment]
Hi，@kuanyshbakytuly，do you have the remake_label() function yet? Could you share it, thanks a lot!

**Comment by tgnrv** [comment]
@Shahtaj-Khalid hello, did you handle the issues with your code? If yes, could share, please? I am also on testing WESAD dataset.

**Comment by tgnrv** [comment]
@kuanyshbakytuly , salam, did you test it properly? Can you share with your code?

**Comment by Texasnum** [comment]
@tgnrv Hello, I'm also running the WESAD dataset and have encountered the problem of remake_label(). If you have solved it, could you share your solution? Thank you!
